packages/syft/src/syft/core/tensor/smpc/share_tensor.py

```diff
@@ -0,0 +1,156 @@
+# stdlib
+from functools import lru_cache
+from typing import Any
+from typing import Optional
+from typing import Tuple
+
+# third party
+from google.protobuf.reflection import GeneratedProtocolMessageType
+import numpy as np
+
+# syft absolute
+from syft.core.common.serde.deserialize import _deserialize as deserialize
+from syft.core.common.serde.serializable import Serializable
+from syft.core.common.serde.serializable import bind_protobuf
+from syft.core.common.serde.serialize import _serialize as serialize
+from syft.core.tensor.passthrough import PassthroughTensor
+from syft.proto.core.tensor.share_tensor_pb2 import ShareTensor as ShareTensor_PB
+
+
+@bind_protobuf
+class ShareTensor(PassthroughTensor, Serializable):
+    def __init__(
+        self,
+        rank: int,
+        ring_size: int = 2 ** 64,
+        value: Optional[Any] = None,
+    ) -> None:
+        self.rank = rank
```

Comment:
Should we raise an exception when rank is not provided or assign a default value?

Reply:
Rank should always be provided. It does not have any default value.